tests/system/Filters/CorsTest.php

```diff
@@ -0,0 +1,461 @@
+ */
+
+namespace CodeIgniter\Filters;
+
+use CodeIgniter\Exceptions\ConfigException;
+use CodeIgniter\HTTP\CLIRequest;
+use CodeIgniter\HTTP\IncomingRequest;
+use CodeIgniter\HTTP\ResponseInterface;
+use CodeIgniter\HTTP\SiteURI;
+use CodeIgniter\HTTP\UserAgent;
+use CodeIgniter\Test\CIUnitTestCase;
+use CodeIgniter\Test\Mock\MockAppConfig;
+use Config\Services;
+
+/**
+ * This test case is based on:
+ *   https://github.com/asm89/stack-cors/blob/master/tests/CorsTest.php
+ *   https://github.com/asm89/stack-cors/blob/b6920bd8996449400ac976e083b55fb45f035467/tests/CorsTest.php
+ *
+ * @internal
+ *
+ * @group Others
+ */
+final class CorsTest extends CIUnitTestCase
+{
+    private Cors $cors;
+    private ResponseInterface $response;
+
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+
+        $this->resetServices();
+    }
+
+    private function createRequest(): IncomingRequest
+    {
+        $config    = new MockAppConfig();
+        $uri       = new SiteURI($config);
+        $userAgent = new UserAgent();
+        $request   = $this->getMockBuilder(IncomingRequest::class)
+            ->setConstructorArgs([$config, $uri, null, $userAgent])
+            ->onlyMethods(['isCLI'])
+            ->getMock();
+        $request->method('isCLI')->willReturn(false);
+
+        return $request;
+    }
+
+    private function createValidPreflightRequest(): IncomingRequest
+    {
+        $request = $this->createRequest();
+
+        return $request->setHeader('Origin', 'http://localhost')
+            ->setHeader('Access-Control-Request-Method', 'GET')
+            ->withMethod('OPTIONS');
+    }
+
+    private function createValidActualRequest(): IncomingRequest
+    {
+        $request = $this->createRequest();
+        $request->setHeader('Origin', 'http://localhost');
+
+        return $request;
+    }
+
+    private function createCors(array $options = [])
+    {
+        $passedOptions = array_merge(
+            [
+                'allowedHeaders'      => ['x-allowed-header', 'x-other-allowed-header'],
+                'allowedMethods'      => ['DELETE', 'GET', 'POST', 'PUT'],
+                'allowedOrigins'      => ['http://localhost'],
+                'exposedHeaders'      => [],
+                'maxAge'              => 0,
+                'supportsCredentials' => false,
+            ],
+            $options
+        );
+
+        return new Cors($passedOptions);
+    }
+
+    public function testBeforeDoesNothingWhenCliRequest(): void
+    {
+        $this->cors = $this->createCors();
+        $cliRequest = new CLIRequest(new MockAppConfig());
+
+        $return = $this->cors->before($cliRequest);
+
+        $this->assertNull($return);
+    }
+
+    private function assertHeader(string $name, string $value): void
+    {
+        $this->assertSame($value, $this->response->getHeaderLine($name));
+    }
+
+    public function testItDoesModifyOnARequestWithoutOrigin()
```

Comment:
I don't know why you didn't use return type `void `in some tests.

Reply:
Just I forgot.

Reply:
Added.